hiring/hiring-process/engineering-hiring/scoring-a-take-home-exercise.md

```diff
@@ -1,53 +1,58 @@
+| Business, presentation, and implementation logic are all mixed together (everything in the Activity/View/Fragment).                                                                                                              | 0     |
+| There is some separation of concerns (ex: separate interface and classes for querying data/making network calls).                                                                                                                | 1     |
+| Candidate has used a standard architecture pattern (MV*/others) or a library (Mobius/MvRx/others). The candidate could also have rolled their own architecture as long as there is clear documentation on their custom solution. | 2     |
+| **Code style**                                                                                                                                                                                                                   |       |
+| Candidate has not followed any particular naming conventions throughout the codebase                                                                                                                                             | 0     |
+| Naming conventions are mostly consistent, with some occasional slips                                                                                                                                                             | 1     |
+| A standard naming convention is followed throughout the codebase                                                                                                                                                                 | 2     |
+| **Code maintainability**                                                                                                                                                                                                         |       |
+| Names are unnecessarily short or meaningless. Ex: `a`, `something`, `doWork`, `runQuery`. An exception to this rule would be something like the iteration variable in a `for` loop.                                              | 0     |
+| Names are mostly descriptive and indicative of the intent. Some names might occasionally be unclear without explanation.                                                                                                         | 1     |
+| Names are mostly descriptive and indicative of the intent. Unclear names have clear comments explaining the purpose of the variable/function and the intent behind it.                                                           | 2     |
+| **Minimum Score Required**                                                                                                                                                                                                       | **6** |
```

Comment:
Are we not including the following in this rubric:
1. DI
2. General programming like use of SOLID principles?
3. How well versed is someone with Kotlin? 
Since there is no negative marking now, we can include some of these things missing as 0 points.

Reply:
I intentionally did not add these because I feel these are things that we will expect people to pick up over time once they start working with us. SOLID principles, for example, are things which most people have never even heard about and these are things that take a lot of time and constant effort to build. I feel something like this is something that should be part of ongoing curriculum and code review rather than being tested for, so I excluded this.

With regards to Kotlin, I again did not add it because whether people are familiar with Kotlin or not is dependent on many factors out of their control. In addition, it is quite easy to get productive with Kotlin within a couple of weeks if you're already familiar with Java and I felt it is unfair to penalise/prefer candidates just because they know Kotlin.